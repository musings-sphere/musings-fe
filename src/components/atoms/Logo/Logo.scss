.main-logo {
	width: 100%;
	height: 100px;
	text-align: center;
	flex-direction: column;
	align-items: center;
	justify-content: center;

	&__image {
		display: flex;
		width: auto;
		height: 70px;
		margin-top: 10px;
		background-size: 36px 36px;
		border-radius: 50%;
		//min-width: 36px;
		align-items: center;
		justify-content: center;
		align-content: center;

		h4 {
			margin-left: 6px;
			font: 500 26px/28px 'Google Sans', Roboto, Helvetica Neue, sans-serif;
			color: rgba(0, 0, 0, 0.8);
			@media (max-width: 539px) {
				//display: none;
			}
		}
	}
}
