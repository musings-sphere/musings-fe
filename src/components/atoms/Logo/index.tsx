import { useRouter } from 'next/router';
import { Stack, Typography } from '@material-ui/core';
import { Image } from '@components/atoms';
import { makeStyles } from '@material-ui/styles';
import { useTheme } from '@material-ui/core/styles';

const logo = 'https://static.almondhydroponics.com/static/logo.png';

const useStyles = makeStyles(() => ({
	logoContainer: {
		cursor: 'pointer',
	},
	container: {
		maxWidth: 'fit-content',
	},
	logoImage: {
		width: '3%',
		height: '3%',
		margin: 0,
		minWidth: 32,
	},
}));

interface Props {
	displayText?: boolean;
}

const Logo = ({ displayText = false }: Props): JSX.Element => {
	const classes = useStyles();
	const router = useRouter();
	const {
		palette: { mode },
	} = useTheme();

	return (
		<div
			className={classes.logoContainer}
			data-testid="logo"
			onClick={() => router.push('/')}
			onKeyDown={() => router.push('/')}
			role="presentation"
		>
			<Stack
				direction="row"
				justifyContent="flex-start"
				alignItems="center"
				spacing={0.5}
				className={classes.container}
			>
				<Image
					className={classes.logoImage}
					src={mode === 'light' ? logo : logo}
					alt="almond"
					lazy={false}
				/>
				{displayText && (
					<Typography
						variant="h5"
						color="textPrimary"
						style={{ fontWeight: 600, fontSize: '16px', padding: '8px' }}
					>
						Musings
					</Typography>
				)}
			</Stack>
		</div>
	);
};

export default Logo;
