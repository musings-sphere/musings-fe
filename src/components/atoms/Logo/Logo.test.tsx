// react library
import { screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
// components
import Logo from './index';
import { ThemeProvider } from '@material-ui/styles';
import getTheme from '../../../theme';

describe.skip('Logo component', () => {
	it('should render correctly', () => {
		const tree = renderer
			.create(
				<ThemeProvider theme={getTheme('light')}>
					<Logo />
				</ThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();

		const elem = screen.getByTestId('logo');
		expect(elem).toHaveClass('makeStyles-logoContainer-1');
	});
});
