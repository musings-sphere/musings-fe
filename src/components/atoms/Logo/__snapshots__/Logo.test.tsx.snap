// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component should render correctly 1`] = `
<div
  className="jss1"
  data-testid="logo"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
>
  <div
    className="jss2 css-vhfwug-Stack-root"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzIiIGhlaWdodD0iMzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt="musings-logo"
          className="image jss4 jss3"
          data-nimg={true}
          decoding="async"
          src="/_next/image?url=https%3A%2F%2Fstatic.almondhydroponics.com%2Fstatic%2Fmusings%2Flogo.png&w=64&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fstatic.almondhydroponics.com%2Fstatic%2Fmusings%2Flogo.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Fstatic.almondhydroponics.com%2Fstatic%2Fmusings%2Flogo.png&w=64&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt="musings-logo"
        className="image jss4 jss3"
        data-nimg={true}
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
